pkg/openstack/instances.go

```diff
@@ -128,14 +130,18 @@ func (i *Instances) NodeAddresses(ctx context.Context, name types.NodeName) ([]v
 // This method will not be called from the node that is requesting this ID. i.e. metadata service
 // and other local methods cannot be used here
 func (i *Instances) NodeAddressesByProviderID(ctx context.Context, providerID string) ([]v1.NodeAddress, error) {
-	klog.V(4).Infof("NodeAddressesByProviderID (%v) called", providerID)
+	klog.V(4).Infof("NodeAddressesByProviderID(%v) called", providerID)
 
-	instanceID, err := instanceIDFromProviderID(providerID)
+	instanceID, instanceRegion, err := instanceIDFromProviderID(providerID)
 
 	if err != nil {
 		return []v1.NodeAddress{}, err
 	}
 
+	if instanceRegion != "" && instanceRegion != i.region {
+		return []v1.NodeAddress{}, nil
```

Comment:
do we need a log for such error?

and return nil means success? should we return error because 
we are getting a provider on different region in such case?

Reply:
Thank you,
I've added logs and error messages where needed. 

Here, I return an empty node-IP list to hide error messages from the main cloud-control module. It does nothing it NodeAddress empty.